Task_Proxy crash if app is closed during export

```python main.py player ~/work/pupil/recordings/surface_tracking_demo/
MainProcess - [INFO] os_utils: Disabled idle sleep.
player - [INFO] launchables.player: Starting new session with '/Users/pabloprietz/work/pupil/recordings/surface_tracking_demo/'
player - [INFO] player_methods: Updating meta info
player - [INFO] video_capture: Install pyrealsense to use the Intel RealSense backend
player - [INFO] calibration_routines.optimization_calibration.build: Building extension modules...
player - [INFO] pupil_detectors.build: Building extension modules...
player - [INFO] plugin: Added: <class 'trigger_matrix_player.Trigger_Matrix'>
player - [INFO] launchables.player: Application Version: 1.0.68
player - [INFO] launchables.player: System Info: User: pabloprietz, Platform: Darwin, Machine: wlan-141-23-165-0.tubit.tu-berlin.de, Release: 17.0.0, Version: Darwin Kernel Version 17.0.0: Thu Aug 24 21:48:19 PDT 2017; root:xnu-4570.1.46~2/RELEASE_X86_64
player - [INFO] camera_models: No user calibration found for camera world at resolution (1280, 720)
player - [INFO] camera_models: No pre-recorded calibration available
player - [WARNING] camera_models: Loading dummy calibration
player - [INFO] fixation_detector: Gaze postions changed. Recalculating.
player - [WARNING] launchables.player: Previous export for range [0-3234] already exsits - overwriting.
player - [WARNING] video_export_launcher: Adding new video export process.
player - [WARNING] video_export_launcher: Added export
player - [INFO] fixation_detector: Created 'fixations.csv' file.
player - [INFO] fixation_detector: Created 'fixation_report.csv' file.
Exception ignored in: <bound method Task_Proxy.__del__ of <background_helper.Task_Proxy object at 0x11a64a2b0>>
Traceback (most recent call last):
  File "/Users/pabloprietz/work/pupil/pupil_src/shared_modules/background_helper.py", line 89, in __del__
    self.cancel(timeout=.1)
  File "/Users/pabloprietz/work/pupil/pupil_src/shared_modules/background_helper.py", line 79, in cancel
    self.process.join(timeout)
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/multiprocessing/process.py", line 119, in join
    assert self._parent_pid == os.getpid(), 'can only join a child process'
AssertionError: can only join a child process
player - [INFO] launchables.player: Process shutting down.
MainProcess - [INFO] os_utils: Re-enabled idle sleep.```

--- a/pupil_src/shared_modules/background_helper.py
+++ b/pupil_src/shared_modules/background_helper.py
@@ -72,11 +72,13 @@ class Task_Proxy(object):
                     yield datum
 
     def cancel(self, timeout=1):
-        self._should_terminate_flag.value = True
-        for x in self.fetch():
-            # fetch to flush pipe to allow process to react to cancel comand.
-            pass
-        self.process.join(timeout)
+        if self.process is not None:
+            self._should_terminate_flag.value = True
+            for x in self.fetch():
+                # fetch to flush pipe to allow process to react to cancel comand.
+                pass
+            self.process.join(timeout)
+            self.process = None
 
     @property
     def completed(self):

